--- pastebinit.orig	2009-03-10 06:24:07.000000000 +0100
+++ pastebinit	2009-07-24 18:59:46.045879618 +0200
@@ -177,6 +177,12 @@
     def getFirstNodeText(nodes, title):
         return getText(getFirstNode(nodes, title).childNodes)
 
+    def nodeExists(nodes, title):
+        node = nodes.getElementsByTagName(title)
+        if node == []:
+            return None
+        return True
+
     # Display usage instructions
     def Usage ():
         print "pastebinit v" + version
@@ -231,10 +237,14 @@
     if gotconfigxml == 1:
         try:
             configxml = xml.dom.minidom.parseString(configtext)
-            website = getFirstNodeText(configxml, "pastebin")
-            user = getFirstNodeText(configxml, "author")
-            format = getFirstNodeText(configxml, "format")
-            jabberid = getFirstNodeText(configxml, "jabberid")
+            if nodeExists(configxml, "pastebin"):
+                website = getFirstNodeText(configxml, "pastebin")
+            if nodeExists(configxml, "author"):
+                user = getFirstNodeText(configxml, "author")
+            if nodeExists(configxml, "format"):
+                format = getFirstNodeText(configxml, "format")
+            if nodeExists(configxml, "jabberid"):
+                jabberid = getFirstNodeText(configxml, "jabberid")
         except KeyboardInterrupt:
             sys.exit(_("KeyboardInterrupt caught."))
         except:
